R/integer64.R

```diff
@@ -1387,74 +1387,106 @@ prod.integer64 <- function(..., na.rm=FALSE) {
   }
 }
 
+all_na_values = function(x, na.rm) {
+  if (length(x) == 0L) return(TRUE)
+  if (!na.rm) return(FALSE)
+  return(all(is.na(x)))
+}
+
 #' @rdname sum.integer64
 #' @export
 min.integer64 = function(..., na.rm=FALSE) {
   l = list(...)
+  na.rm = isTRUE(na.rm)
+  ret = NULL
+  
   if (length(l) == 1L) {
-    ret = .Call(C_min_integer64, l[[1L]], na.rm, double(1L))
-    oldClass(ret) = "integer64"
+    if (length(l[[1]]) > 0L) {
+      ret = .Call(C_min_integer64, l[[1L]], na.rm, double(1L))
+      oldClass(ret) = "integer64"
+    }
   } else {
-    ret = vapply(l, FUN.VALUE=integer64(1L), function(e) {
+    ret = vapply(Filter(length, l), FUN.VALUE=integer64(1L), function(e) {
       if (is.integer64(e)) {
         .Call(C_min_integer64, e, na.rm, double(1L))
       } else {
-        as.integer64(min(e, na.rm=na.rm))
+        suppressWarnings(as.integer64(min(e, na.rm=na.rm)))
       }
     })
     oldClass(ret) = "integer64"
-    ret = min(ret, na.rm=na.rm)
+    if (!all_na_values(ret, na.rm))
```

Comment:
this might mean we have to check `all_na_values()` twice, right?

how about just deferring this to below:

```
if (all_na_values(...)) {
  ...
} else {
  ret = min(ret, na.rm=na.rm)
}
```

WDYT?